internal/app/agent/service.go

```diff
@@ -62,21 +71,27 @@ type Service interface {
 	ServiceConfig(uuid, cmdStr string) error
 
 	// Services returns service list
-	Services() map[string]*services.Service
+	Services() []ServiceInfo
 
 	// Publish message
 	Publish(string, string) error
 }
 
 var _ Service = (*agent)(nil)
 
+type ServiceInfo struct {
+	Name     string
+	LastSeen time.Time
+	Status   string
+}
+
 type agent struct {
 	mqttClient  paho.Client
 	config      *config.Config
 	edgexClient edgex.Client
 	logger      log.Logger
 	nats        *nats.Conn
-	servs       map[string]*services.Service
+	svcs        map[string]*services.Service
```

Comment:
Why is this a pointer? Do we need to changed by reference somewhere? Can map of simple simple `services.Service{}` values be used?

Reply:
there is an issue created for this https://github.com/mainflux/agent/issues/20